src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -794,6 +870,20 @@ public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, R
             return resource;
         }
 
+        /// <summary>
+        /// Creates the <see cref="ODataDeletedResource"/> to be written while writing this resource.
+        /// </summary>
+        /// <param name="selectExpandNode">The <see cref="SelectExpandNode"/> describing the response graph.</param>
+        /// <param name="resourceContext">The context for the resource instance being written.</param>
+        /// <returns>The created <see cref="ODataDeletedResource"/>.</returns>
+        public virtual ODataDeletedResource CreateDeletedResource(SelectExpandNode selectExpandNode, ResourceContext resourceContext)
```

Comment:
I think you should move this method to the section where other public methods are